### PR TITLE
write parsing datetime tests

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 1]
```diff
@@ -1774,24 +1762,554 @@ abstract class StdLibSpec extends Qspec {
...
-      "OffsetDateTime" >> prop { (v: JOffsetDateTime) =>
-        unary(OffsetDateTime(_).embed, Data.Str(v.toString), Data.OffsetDateTime(v))
+      "OffsetDateTime" >> {
+        "pre-Gregorian format" >> {
+          val expected = Data.OffsetDateTime(JOffsetDateTime.of(JLocalDateTime.of(JLocalDate.of(53, 4, 2), JLocalTime.of(7, 47, 18)), ZoneOffset.of("+02:30")))
+
+          unary(OffsetDateTime(_).embed, Data.Str("0053-04-02T07:47:18+02:30"), expected)
+          unary(OffsetDateTime(_).embed, Data.Str("0053-04-0207:47:18+02:30"), expected)
+          unary(OffsetDateTime(_).embed, Data.Str("00530402T074718+0230"), expected)
+          unary(OffsetDateTime(_).embed, Data.Str("00530402074718+0230"), Data.NA) // FIXME this should parse
```
Comment (alissapajer) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala:1772]: Ideally we could document the entire ISO 8601 specification with these tests and then skip them when we don't conform to spec. But we can't do that in this test framework because of how it's separated from `mimir`.

I suppose I could `skipped` the test _here_, maybe that's better than a `FIXME`? Then we could add more `skipped` tests here as documentation of exactly what parts of the spec we don't yet support.

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 2]
```diff
@@ -98,35 +98,23 @@ abstract class StdLibSpec extends Qspec {
...
 
     implicit val arbOffsetDateTime: Arbitrary[JOffsetDateTime] =
       Arbitrary[JOffsetDateTime]((arbLocalDateTime.gen, runner.timezoneDomain) >> JOffsetDateTime.of)
-    implicit val arbDataOffsetDateTime: Arbitrary[Data.OffsetDateTime] =
-      arbOffsetDateTime ^^ Data.OffsetDateTime
 
     implicit val arbInterval: Arbitrary[DateTimeInterval] =
       Arbitrary[DateTimeInterval](runner.intervalDomain)
-    implicit val arbDataInterval: Arbitrary[Data.Interval] =
-      arbInterval ^^ Data.Interval
```
Comment (alissapajer) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala:129]: All of the removed implicits were not used.

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 3]
```diff
@@ -1774,24 +1762,554 @@ abstract class StdLibSpec extends Qspec {
...
+
+        "undefined when offset sign is missing" >> {
+          unary(OffsetTime(_).embed, Data.Str("07:47:18,04159305:00"), Data.NA)
+          unary(OffsetTime(_).embed, Data.Str("07:47:18,04159305"), Data.NA)
+          unary(OffsetTime(_).embed, Data.Str("074718,0415930500"), Data.NA)
+          unary(OffsetTime(_).embed, Data.Str("074718,04159305"), Data.NA)
+        }
+
+        "arbitrary" >> prop { (v: JOffsetTime) =>
+          unary(OffsetTime(_).embed, Data.Str(v.toString), Data.OffsetTime(v))
```
Comment (alissapajer) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala:2150]: Previously all we had for tests were these scalacheck tests. I don't actually expect them to catch any bugs because they are only checking against a single format (what `.toString` on the datetime prints). But I figured there's no harm in keeping them around as long as we supplement with tests encoding actual corner cases.

**Comment by alissapajer** [comment]
@djspiewak could I get a review on this, as well as its mimir counterpart https://github.com/precog/mimir/pull/904
